tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -167,7 +198,21 @@ public async Task Sanitize(IEnumerable<RecordedTestSanitizer> sanitizers, bool s
             {
                 foreach (var sanitizer in sanitizers)
                 {
+                    var entriesPreSanitize = Array.Empty<RecordEntry>();
+                    if (DebugLogger.CheckLogLevel(LogLevel.Debug))
+                    {
+                        entriesPreSanitize = this.Entries.Select(requestEntry => { requestEntry.initializeIsModifiedFlag(); return CopyRecordEntry(requestEntry); }).ToArray();
```

Comment:
Same cleanup if `InitializeIsModifiedFlag`. Should be possible in defaults. Then we should just `return requestEntry.Clone()`

Reply:
calling it `Reset..` now, but still keeping it as discussed offline.